x/stake/keeper/delegation.go

```diff
@@ -129,6 +149,26 @@ func (k Keeper) GetRedelegation(ctx sdk.Context,
 	return red, true
 }
 
+// load all redelegations from a particular validator
+func (k Keeper) GetRedelegationsFromValidator(ctx sdk.Context, valAddr sdk.Address) (redelegations []types.Redelegation) {
```

Comment:
same as previous - maybe this should take a function to operate on each rebonding delegation?

Reply:
See above comment/question.